From 48c3b095e788993e1570f87a979c3bdcc0fe86a2 Mon Sep 17 00:00:00 2001
From: David Ward <david.ward@ll.mit.edu>
Date: Sun, 5 Apr 2026 15:33:16 -0400
Subject: [PATCH 2/5] lib: Remove lyd_new_ext_term(), lyd_change_term_bin()
 wrappers

These functions were removed in libyang 4.8.0 and 5.3.0 respectively.
They are not called by FRR.

Signed-off-by: David Ward <david.ward@ll.mit.edu>
Signed-off-by: Daniel Baumann <mail@daniel-baumann.ch>
---
 lib/yang.c | 14 --------------
 lib/yang.h | 12 ------------
 2 files changed, 26 deletions(-)

diff --git a/lib/yang.c b/lib/yang.c
index 09c99dc996..66f722f7b0 100644
--- a/lib/yang.c
+++ b/lib/yang.c
@@ -1717,9 +1717,7 @@ LY_ERR yang_lyd_parse_data(const struct ly_ctx *ctx, struct lyd_node *parent,
  */
 
 #undef lyd_new_term_bin
-#undef lyd_change_term_bin
 #undef lyd_new_path2
-#undef lyd_new_ext_term
 
 #if (LY_VERSION_MAJOR >= 4)
 #define LY_SZ(x) ((x)*8)
@@ -1734,11 +1732,6 @@ LY_ERR yang_new_term_bin(struct lyd_node *parent, const struct lys_module *modul
 	return lyd_new_term_bin(parent, module, name, value, LY_SZ(size), options, node);
 }
 
-LY_ERR yang_change_term_bin(struct lyd_node *term, const void *value, uint32_t size)
-{
-	return lyd_change_term_bin(term, value, LY_SZ(size));
-}
-
 LY_ERR yang_new_path2(struct lyd_node *parent, const struct ly_ctx *ctx, const char *path,
 		      const void *value, uint32_t size, LYD_ANYDATA_VALUETYPE value_type,
 		      uint32_t options, struct lyd_node **new_parent, struct lyd_node **new_node)
@@ -1747,11 +1740,4 @@ LY_ERR yang_new_path2(struct lyd_node *parent, const struct ly_ctx *ctx, const c
 			     new_parent, new_node);
 }
 
-#if (LY_VERSION_MAJOR >= 3) && (LY_VERSION_MAJOR < 5)
-LY_ERR yang_new_ext_term(const struct lysc_ext_instance *ext, const char *name, const void *value,
-			 uint32_t size, uint32_t options, struct lyd_node **node)
-{
-	return lyd_new_ext_term(ext, name, value, LY_SZ(size), options, node);
-}
-#endif
 #undef LY_SZ
diff --git a/lib/yang.h b/lib/yang.h
index 2dd2934072..2764647d5f 100644
--- a/lib/yang.h
+++ b/lib/yang.h
@@ -957,24 +957,12 @@ extern LY_ERR yang_new_term_bin(struct lyd_node *parent, const struct lys_module
 #define lyd_new_term_bin(...)                                                                     \
 	_Pragma("GCC error \"Use yang_new_term_bin() instead of lyd_new_term_bin()\"")
 
-LY_ERR yang_change_term_bin(struct lyd_node *term, const void *value, uint32_t value_size_bits);
-#define lyd_change_term_bin(...)                                                                  \
-	_Pragma("GCC error \"Use yang_change_term_bin() instead of lyd_new_change_term_bin()\"")
-
 extern LY_ERR yang_new_path2(struct lyd_node *parent, const struct ly_ctx *ctx, const char *path,
 			     const void *value, uint32_t value_size_bits,
 			     LYD_ANYDATA_VALUETYPE value_type, uint32_t options,
 			     struct lyd_node **new_parent, struct lyd_node **new_node);
 #define lyd_new_path2(...) _Pragma("GCC error \"Use yang_new_path2() instead of lyd_new_path2()\"")
 
-#if (LY_VERSION_MAJOR >= 3) && (LY_VERSION_MAJOR < 5)
-extern LY_ERR yang_new_ext_term(const struct lysc_ext_instance *ext, const char *name,
-				const void *value, uint32_t value_size_bits, uint32_t options,
-				struct lyd_node **node);
-#define lyd_new_ext_term(...)                                                                     \
-	_Pragma("GCC error \"Use yang_new_ext_term() instead of lyd_new_ext_term()\"")
-#endif
-
 #ifdef __cplusplus
 }
 #endif
-- 
2.55.0

